exchange/exchange_test.go

```diff
@@ -507,6 +507,516 @@ func TestTwoBiddersDebugDisabledAndEnabled(t *testing.T) {
+			},
+		},
+		{
+			"customRates empty, UsePBSRates set to false. Because pbsRates cannot be used, we return constant rates",
+			testInput{
+				pbsRates: pbsRates,
+				bidExtCurrency: &openrtb_ext.ExtRequestCurrency{
+					// ConversionRates inCustomRates not initialized makes for a zero-length map
+					UsePBSRates: &boolFalse,
+				},
+			},
+			testOutput{
+				constantRates: true,
+			},
+		},
+		{
+			"customRates is nil, pbsRates are not nil. Resulting rates will be identical to pbsRates",
+			testInput{
+				pbsRates:       pbsRates,
+				bidExtCurrency: nil,
+			},
+			testOutput{
+				resultingRates: pbsRates,
+			},
+		},
+		{
+			"customRates empty, UsePBSRates set to false, pbsRates are nil. Return default currency converter where we only have USDs",
+			testInput{
+				pbsRates: nil,
+				bidExtCurrency: &openrtb_ext.ExtRequestCurrency{
+					// ConversionRates inCustomRates not initialized makes for a zero-length map
+					UsePBSRates: &boolFalse,
+				},
+			},
+			testOutput{
+				constantRates: true,
+			},
+		},
+		{
+			"customRates empty, UsePBSRates set to true, pbsRates are nil. Return default currency converter where we only have USDs",
+			testInput{
+				pbsRates: nil,
+				bidExtCurrency: &openrtb_ext.ExtRequestCurrency{
+					// ConversionRates inCustomRates not initialized makes for a zero-length map
+					UsePBSRates: &boolFalse,
+				},
+			},
+			testOutput{
+				constantRates: true,
+			},
+		},
+		{
+			"customRates empty, UsePBSRates set to true, pbsRates are nil. Return default constant rates converter",
+			testInput{
+				pbsRates: nil,
+				bidExtCurrency: &openrtb_ext.ExtRequestCurrency{
+					// ConversionRates inCustomRates not initialized makes for a zero-length map
+					UsePBSRates: &boolTrue,
+				},
+			},
+			testOutput{
+				constantRates: true,
+			},
+		},
+		{
+			"customRates is nil, pbsRates are nil. Return default constant rates converter",
+			testInput{
+				pbsRates:       nil,
+				bidExtCurrency: nil,
+			},
+			testOutput{
+				constantRates: true,
+			},
+		},
+	}
+
+	for _, tc := range testCases {
+
+		// Test setup:
+		jsonPbsRates, err := json.Marshal(tc.given.pbsRates)
+		if err != nil {
+			t.Fatalf("Failed to marshal PBS rates: %v", err)
+		}
+
+		// Init mock currency conversion service
+		mockCurrencyClient := &mockCurrencyRatesClient{
+			responseBody: `{"dataAsOf":"2018-09-12","conversions":` + string(jsonPbsRates) + `}`,
+		}
+		mockCurrencyConverter := currency.NewRateConverter(
+			mockCurrencyClient,
+			"currency.fake.com",
+			24*time.Hour,
+		)
+		mockCurrencyConverter.Run()
+
+		e := new(exchange)
+		e.currencyConverter = mockCurrencyConverter
+
+		// Run test
+		auctionRates := e.getAuctionCurrencyRates(tc.given.bidExtCurrency)
+
+		// When fromCurrency and toCurrency are the same, a rate of 1.00 is always expected
+		rate, err := auctionRates.GetRate("USD", "USD")
+		assert.NoError(t, err, tc.desc)
+		assert.Equal(t, rate, rate, tc.desc)
```

Comment:
Shouldn't the expected rate here be `1`?

Reply:
Yes. Good catch